client/src/views/devices/ImportRecoveryDevicePage.vue

```
@@ -0,0 +1,329 @@
+        notificationManager.showToast(
+          new Notification({
+            title: t('ImportRecoveryDevicePage.errors.fileErrorTitle'),
+            message: t('ImportRecoveryDevicePage.errors.fileErrorMessage'),
+            level: NotificationLevel.Error,
+          }),
+        );
+        break;
+      case RecoveryImportErrorTag.Internal:
+        notificationManager.showToast(
+          new Notification({
+            title: t('ImportRecoveryDevicePage.errors.internalErrorTitle'),
+            message: t('ImportRecoveryDevicePage.errors.internalErrorMessage'),
+            level: NotificationLevel.Error,
+          }),
+        );
+        break;
+    }
+  }
+}
+
+async function createNewDevice(): Promise<void> {
+  // Check matching and valid passwords
+  if (!choosePasswordInput.value || !(await choosePasswordInput.value.areFieldsCorrect())) {
+    notificationManager.showToast(
+      new Notification({
+        title: t('ImportRecoveryDevicePage.errors.passwordErrorTitle'),
+        message: t('ImportRecoveryDevicePage.errors.passwordErrorMessage'),
+        level: NotificationLevel.Error,
+      }),
+    );
+    return;
+  }
+  // Check new device info exists
+  if (!newDeviceInfo.value) {
+    notificationManager.showToast(
+      new Notification({
+        title: t('ImportRecoveryDevicePage.errors.internalErrorTitle'),
+        message: t('ImportRecoveryDevicePage.errors.internalErrorMessage'),
+        level: NotificationLevel.Error,
+      }),
+    );
+    return;
+  }
+  // Save new device with password
+  if (!(await saveDevice(newDeviceInfo.value, 'newPassword')).ok) {
+    notificationManager.showToast(
+      new Notification({
+        title: t('ImportRecoveryDevicePage.errors.saveDeviceErrorTitle'),
+        message: t('ImportRecoveryDevicePage.errors.saveDeviceErrorMessage'),
+        level: NotificationLevel.Error,
+      }),
+    );
+    return;
+  }
+  // Delete previous device
+  if (!(await deleteDevice(props.device)).ok) {
```

You can fail silently in this case.

```typescript
await deleteDevice(props.device);
state.value = ImportDevicePageState.Done;
```